File: src/main/java/TendersPage.java
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

import java.util.List;
import java.util.stream.Collectors;

import static org.testng.Assert.assertTrue;

public class TendersPage extends BasePage {

 //   List<WebElement> repositoriesList = driver.findElements(By.xpath("//td[text() = 'AutoTest Public Tender']"));

    private By publicTenderLink = By.xpath("//td[text() = 'AutoTest Public Tender']");
    private final static String TITLE = "Tenders";

    public TendersPage(WebDriver driver) {
        super(driver);
    }

    public TendersPage openPublicTender(){
        driver.findElement(publicTenderLink).click();
        return this;
        }
    }

//    public List<String> getRepositories() {
//        List<String> repositList = repositoriesList.stream().map(rep -> rep.getText()).collect(Collectors.toList());
//        return repositList;
//    }
//}

File: src/main/java/main/TendersPage.kt
package main

import BasePage
import org.openqa.selenium.By
import org.openqa.selenium.WebDriver

public class TendersPage(driver: WebDriver?) : BasePage(driver) {
    private val publicTenderLink: By = By.xpath("//td[text() = 'AutoTest Public Tender']")
    private val limitedTenderLink: By = By.xpath("//td[text() = 'Limited 114']")
    private val browserFrame: By = By.xpath("//frame[@name='browser']")
    private val functionsFrame: By = By.xpath("//frame[@name='functions']")
    private val navigationFrame: By = By.xpath("//frame[@name='navigation']")
    private val directoryFrame: By = By.xpath("//frame[@name='directory']")
    private val fileDetailsFrame: By = By.xpath("//frame[@name='filespec']")

    fun openPublicTender(): TendersPage {
        driver?.findElement(limitedTenderLink)?.click()
        return this
    }

    fun openLimitedTender(): TendersPage {
        driver?.findElement(limitedTenderLink)?.click()
        return this
    }

    fun switchToBrowserFrame() {
        driver?.switchTo()?.frame(driver?.findElement(browserFrame))
    }

    fun switchToFunctionsFrame() {
        driver?.switchTo()?.frame(driver?.findElement(functionsFrame))
    }

    fun switchToNavigationFrame() {
        driver?.switchTo()?.frame(driver?.findElement(navigationFrame))
    }

    fun switchToDirectoryFrame() {
        driver?.switchTo()?.frame(driver?.findElement(directoryFrame))
    }

    fun switchToFileDetailsFrame() {
        driver?.switchTo()?.frame(driver?.findElement(fileDetailsFrame))
    }

    fun switchToPreviousWindow() {
        val windowHandles = driver?.windowHandles

        // Якщо є більше одного вікна, то можна повернутися до попереднього
        if (windowHandles?.size ?: 0 > 1) {
            val iterator: Iterator<String> = windowHandles!!.iterator()
            val currentWindow = iterator.next()
            val previousWindow = iterator.next()

            driver?.switchTo()?.window(previousWindow)
        } else {
            println("No previous window")
        }
    }
}
